### PR TITLE
Add span information to logger after span creation

### PR DESCRIPTION
**Pull request by hauleth**
Close #1

### CODE REVIEW

**Comment by seancribbs** [comment]
LGTM 👍

**Comment by seancribbs** [comment]
Oops, looks like you need to capture the header name too in the `call` function:

```== Compilation error in file lib/knapsack/tracing_plug.ex ==
** (CompileError) lib/opencensus/plug/trace.ex:94: undefined function header/0
    (stdlib) lists.erl:1338: :lists.foreach/2
    (stdlib) erl_eval.erl:680: :erl_eval.do_apply/6
    (elixir) lib/kernel/parallel_compiler.ex:198: anonymous fn/4 in Kernel.ParallelCompiler.spawn_workers/6
```

BTW I added `location: :keep` to the `quote` to get the right line number above.

File: lib/opencensus/plug/trace.ex [hunk 1]
```diff
@@ -111,14 +120,5 @@ defmodule Opencensus.Plug.Trace do
          ctx when ctx != :undefined <- :oc_span_ctx_header.decode(val) do
       :ocp.with_span_ctx(ctx)
     end
-
```
Comment (seancribbs) [inline lib/opencensus/plug/trace.ex]: This function no longer returns `conn`, and so breaks the `call` function above (first line of the function). Its return value should probably be ignored.

File: lib/opencensus/plug/trace.ex [hunk 2]
```diff
@@ -83,7 +83,16 @@ defmodule Opencensus.Plug.Trace do
...
-        Plug.Conn.register_before_send(conn, fn conn ->
+        encoded =
+          :ocp.current_span_ctx()
+          |> :oc_span_ctx_header.encode()
+          |> :erlang.iolist_to_binary()
+
+        Logger.metadata(tracespan: encoded)
+
+        conn
+        |> Plug.Conn.put_resp_header(header, encoded)
```
Comment (seancribbs) [inline lib/opencensus/plug/trace.ex]: `header` is unbound here, and also is expected to be lowercase in Plug.
```suggestion
        |> Plug.Conn.put_resp_header(String.downcase(header), encoded)
```

**Review by seancribbs — COMMENTED** [review]
Just left comments on the bugs I found in this.

File: lib/opencensus/plug/trace.ex (hunk 1 above)
Comment (hauleth) [reply lib/opencensus/plug/trace.ex]: Fixed, as this no longer needs to return `conn`.

**Review by seancribbs — APPROVED** [review]
This works now! 💯

**Comment by tsloughter** [comment]
I don't know how yet but it would be good if the logger metadata could be set in opencensus lib instead of in here. So it worked for any Elixir app.

**Comment by hauleth** [comment]
@tsloughter ideally Elixir's `Logger` would reuse Erlang `logger` metadata. I am working on it in the core.

**Comment by tsloughter** [comment]
Ah, very nice.

**Review by tsloughter — APPROVED** [review]

**Comment by hauleth** [comment]
elixir-lang/elixir#8515 for reference. I would like to provide Elixir logger implementation to be thin wrapper over Erlang `logger` in case when it is available (problem is that Elixir needs to support OTP 19+ at least until 1.10).

**Pull request merged**